src/safeds/data/tabular/transformation/_functional_table_transformer.py

```diff
@@ -0,0 +1,118 @@
+    from collections.abc import Callable
+
+    from safeds.data.tabular.containers import Table
+
+from ._table_transformer import TableTransformer
+
+
+class FunctionalTableTransformer(TableTransformer):
+    """
+    Learn a transformation for a set of columns in a `Table` and transform another `Table` with the same columns.
+
+    Parameters
+    ----------
+    funct:
+        The Callable that receives a table and returns a table that is to be wrapped.
+    """
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Dunder methods
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def __init__(
+        self,
+        funct: Callable[[Table], Table],
+    ) -> None:
+        super().__init__(None)
+        self._func = funct
+
+    def __hash__(self) -> int:
+        return _structural_hash(
+            super().__hash__(),
+            self._func,
+        )
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Properties
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @property
+    def is_fitted(self) -> bool:
+        """FunctionalTableTransformer is always considered to be fitted."""
+        return True
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Learning and transformation
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def fit(self, table: Table) -> FunctionalTableTransformer:  # noqa: ARG002
+        """
+        **Note:** For FunctionalTableTransformer this is a no-OP.
+
+        Parameters
+        ----------
+        table:
+            Required only to be consistent with other transformers.
+
+        Returns
+        -------
+        fitted_transformer:
+            Returns self, because this transformer is always fitted.
+
+        """
+        return self
+
+    def transform(self, table: Table) -> Table:
+        """
+        Apply the callable to a table.
+
+        **Note:** The given table is not modified.
+
+        Parameters
+        ----------
+        table:
+            The table on which on which the callable is executed.
+
+        Returns
+        -------
+        transformed_table:
+            The transformed table.
+
+        Raises
+        ------
+        Exception:
+            Raised when the wrapped callable encounters an error.
+
+        """
+        try:
+            return self._func(table)
+        except Exception as e:
+            # TODO Evaluate if switch to non-generic exception is useful, as _func can be any callable
+            raise Exception("The underlying function encountered an error") from e  # noqa: TRY002
```

Comment:
Would also be fine to remove the wrapping into another `Exception` completely and just let the original exception bubble up. This addtional exception does not add useful information.

Reply:
Wrapper has been removed